src/data/set/prod.lean

```diff
@@ -309,8 +313,8 @@ lemma mem_diagonal (x : α) : (x, x) ∈ diagonal α := by simp [diagonal]
 
 @[simp] lemma mem_diagonal_iff {x : α × α} : x ∈ diagonal α ↔ x.1 = x.2 := iff.rfl
 
-instance decidable_mem_diagonal [h : decidable_eq α] (x : α × α) : decidable (x ∈ diagonal α) :=
-h x.1 x.2
+instance decidable_mem_diagonal [h : decidable_eq α] : decidable_pred (∈ diagonal α) :=
+λ x, h x.1 x.2
```

Comment:
Note that nothing was incorrect here; `decidable (x ∈ s)` and `decidable_pred (∈ s)` are both fine, it's `decidable_pred s` that is bad.

Reply:
Sure, there had been some discussion when I had the PR for this instance as well, but generally sticking with `∈` makes sense then.